toke.c

```diff
@@ -4590,10 +4590,7 @@ S_intuit_more(pTHX_ char *s, char *e,
      * written, and regcurly never required a comma, as in {0}.  Probably it is
      * ok as-is */
     if (s[0] == '{') {
-        if (regcurly(s, e, NULL)) {
-            return FALSE;
-        }
-        return TRUE;
+        return ! regcurly(s, e, NULL);
```

Comment:
Great change. Former pattern appears to be used in other places, marking such cleanup as my next TODO once my current PR is merged.